allennlp/models/encoder_decoders/wikitables_semantic_parser.py

```diff
@@ -271,6 +310,166 @@ def forward(self,  # type: ignore
             # TODO(matt): compute accuracy here.
             return outputs
 
+    @staticmethod
+    def _get_neighbor_indexes(worlds: List[WikiTablesWorld],
+                              num_entities: int,
+                              tensor: torch.Tensor) -> Variable:
```

Comment:
The distinction between `Variable` and `Tensor` in pytorch is a bit awkward, and is supposed to be going away in pytorch 0.4.  We've been annotating types with `torch.Tensor` instead of `Variable`, partially looking ahead to the type change that's coming.  So, unless mypy complains, I'd switch everywhere that you have `Variable` annotations to `torch.Tensor` annotations (and if you can be more specific, like `torch.LongTensor` or `torch.FloatTensor`, even better).

Reply:
The difference here is that the _input_ tensor probably needs to be annotated as a `Variable`, because you're actually passing it a variable.  I'm actually surprised that mypy isn't complaining here, because `torch.Tensor` doesn't have a `data` member...  It'll be really nice when pytorch unifies these two types.